### ISSUE TITLE
Проблема с загрузкой парсеров Android 14

### ISSUE THREAD

**Issue by theoneklinch** [veopot/mwx-json#84]
Доброго времени суток.

При загрузке парсеров из репозитория выводится ошибка о необходимости создать папку "Parsers".

![Screenshot_20240418-192717](https://github.com/veopot/mwx-json/assets/167519524/181ed5a9-7937-4769-9bd3-57084818d3ef)

Проблема в том, что папка уже создана самим приложением и в ней даже находится файл repo.ejson

![Screenshot+2024-04-18+1944192](https://github.com/veopot/mwx-json/assets/167519524/be4b2171-5ff6-417a-a691-1b43ea2a067a)

Немножко странно, что приложение не видит собственную папку, и есть теория, что оно пытается скачать парсер, куда то не в то место, но окно ошибки слишком мало, чтобы оценить путь, а файла с логами я не нашел, чтобы просмотреть ошибку полностью. 
Все необходимые разрешения в системе для приложения я предоставил

Устройство: Google pixel 7 
ОС: Android 14
Версия приложения: 3.1.6.8 (та, что для а14)

Так как начиная, кажется, с 11 андроида, системой ужесточен доступ пользователя к Android/Data , может быть, есть смысл  вывести часть каталогов, в том числе с парсерами, из под android/data куда-нибудь в корень sdcard (имею ввиду внутреннюю память устройства), как, например, сделали идейные подражатели (не буду указывать пальцем, но заканчивается на слово лабораторияХ). Туда, например, и у пользователя будет свободный доступ, чтобы что то доливать.

**Comment by Moris-Shalon** [comment]
Есть информация о решении данной проблемы на Android 14? Только из-за приложения MangaWatcherX я ещё не обновился, а надо бы, т.к. на новой прошивке исправили часть косяков моего телефона.

**Comment by Moris-Shalon** [comment]
@LomikXP, есть какая-нибудь информация на этот счёт? Лишь этот нюанс и удерживает меня сейчас от обновления до Android 14, на котором часть багов моего телефона устранили

**Comment by Moris-Shalon** [comment]
известно что-нибудь об этом? Несколько месяцев как не обновляюсь на Android 14 из-за одной лишь этой проблемы. Проверить самостоятельно, проявится ли она на моём смартфоне, я смогу только после обновления. Но, если я обновлюсь, откатиться обратно уже не смогу. Сижу и жду смиренно инфы.

**Comment by Moris-Shalon** [comment]
@LomikXP, известно что-нибудь об этом? Уже месяца 2 или 3 как не обновляюсь на Android 14 из-за одной лишь этой проблемы. Проверить самостоятельно, проявится ли она на моём смартфоне, я смогу только после обновления. Но, если я обновлюсь, откатиться обратно уже не смогу. Фикс под Android 14, хоть, в работе?

**Comment by LomikXP** [comment]
> @LomikXP, известно что-нибудь об этом? Уже месяца 2 или 3 как не обновляюсь на Android 14 из-за одной лишь этой проблемы. Проверить самостоятельно, проявится ли она на моём смартфоне, я смогу только после обновления. Но, если я обновлюсь, откатиться обратно уже не смогу. Фикс под Android 14, хоть, в работе?

Нужны подробности, так как у меня не повторяется на эмуляторе Pixel XL с Android 14
1. Какая версия приложения (была еще одна версия для api23, лежит здесь в репо)?
2. Приложение проверяется с нуля (то есть новое устройство, просто установили приложение и сразу ошибка)?
3. Какая последовательность действий?
4. На всех парсерах проблема?
5. Отправьте лог сразу после ошибки (долгий там по пункту в настройках О программе - История изменений)

**Comment by Moris-Shalon** [comment]
Так, поскольку получил запрос на логи от @LomikXP, я обновился до Android 14, чтобы самому столкнуться с этим багом и предоставить логи. Однако я не столкнулся с этой проблемой на своём Asus Rog Phone 6. Всё ок, старые парсеры работают, новые также добавляются, манга старая апдейтится, новая добавляется, главы загружаются. Единственный нюанс - приложение должно метить на Android 6.0+, чтобы его установить можно было. Но приложения, что уже были установлены до апдейта до Android 14, как работали, так и работают.
Короче, у меня лично ок. Зря апдейт откладывал на 3 месяца.

Данные тогда надо запрашивать от автора Issue, @theoneklinch

**Comment by LomikXP** [comment]
> Так, поскольку получил запрос на логи от @LomikXP, я обновился до Android 14, чтобы самому столкнуться с этим багом и предоставить логи. Однако я не столкнулся с этой проблемой на своём Asus Rog Phone 6. Всё ок, старые парсеры работают, новые также добавляются, манга старая апдейтится, новая добавляется, главы загружаются. Единственный нюанс - приложение должно метить на Android 6.0+, чтобы его установить можно было. Но приложения, что уже были установлены до апдейта до Android 14, как работали, так и работают. Короче, у меня лично ок. Зря апдейт откладывал на 3 месяца.
> 
> Данные тогда надо запрашивать от автора Issue, @theoneklinch

Есть версия с приставкой api23, но на нее тоже чот ругается, пока не разбирался из-за чего, это уже другая история

**Comment by LomikXP** [comment]
@theoneklinch тот же вопрос к вам, я не исключаю текущий баг, вполне реальная проблема, нужны ответы на вопросы

**Comment by veopot** [comment]
Была еще одна девушка но проблему решили через adb 
Создав папку парсер и вручную закинув

**Comment by LomikXP** [comment]
> Была еще одна девушка но проблему решили через adb Создав папку парсер и вручную закинув

Ну это не общее решение, мне повторить бы
